src/WebUI/Components/KubeSummary.tsx

```diff
@@ -255,6 +283,112 @@ export class KubeSummary extends BaseComponent<IKubeSummaryProps, IKubernetesCon
+        this.setState({
+            workloadsFilterState: this.state.filter.getState()
+        })
+    };
+
+    private _onSvcFilterApplied = (currentState: IFilterState) => {
+        this.setState({
+            svcFilterState: this.state.svcFilter.getState()
+        })
+    };
+
+    private _getNameFilterKey(): string | undefined {
+        const filterState:IFilterState | undefined = this.state.selectedKey===workloadsPivotItemKey?this.state.workloadsFilterState:this.state.svcFilterState;
+        const filterItem: IFilterItemState | null= filterState? filterState["nameKey"]: null;
+        return filterItem? (filterItem.value as string): undefined;
+    }
+
+    private _getTypeSelection():any[] {
+        const filterState:IFilterState | undefined = this.state.selectedKey===workloadsPivotItemKey?this.state.workloadsFilterState:this.state.svcFilterState;
+        const filterItem: IFilterItemState | null= filterState? filterState["typeKey"]: null;
+        const selections: any[] = filterItem? filterItem.value : [];
+        return selections;
+    }
+
+    private _showComponent(resourceType: KubeResourceType): boolean{
+        const selections: KubeResourceType[] = this._getTypeSelection();
+        // if no selections are made, show all components
+        if(selections.length > 0) {
+            return selections.indexOf(resourceType) != -1;
+        }
+        return true;
+    }
+
+    private _pickListItems = () => {
+        return [KubeResourceType.Deployments, 
+                KubeResourceType.ReplicaSets, 
+                KubeResourceType.DaemonSets, 
+                KubeResourceType.StatefulSets, 
+                KubeResourceType.Pods];
+    };
+
+    private _listItems = (item: any)=> {
+        let name:string = "";
+        switch(item){
+            case KubeResourceType.Deployments:
+                name = Resources.DeploymentsDetailsText;
+                break;
+            case KubeResourceType.ReplicaSets:
+                name = Resources.ReplicaSetText;
+                break;
+            case KubeResourceType.DaemonSets:
+                name = Resources.DaemonSetText;
+                break;
+            case KubeResourceType.StatefulSets:
+                name = Resources.StatefulSetText;
+                break;
+            case KubeResourceType.Pods:
+                name = Resources.PodsText;
+                break;
+            default:
+                name = "Unknown";
+                break;
+     };
+        return {
+            key: item.toString(),
+            name: name
+        };
+    };
+
+    private _generateSvcTypes(): string[] {
+        let svcTypes: string[] = [];
+        this.state.serviceList && this.state.serviceList.items && this.state.serviceList.items.forEach((svc) => {
+            if (svcTypes.indexOf(svc.spec.type) == -1) {
+                svcTypes.push(svc.spec.type);
+            }
+        });
+        return svcTypes;

```

Comment:
did we check if service types are zero what happens?

Reply:
Services will be rendered unfiltered